### PR TITLE
DOC: merge hatch demos

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Following discussion in #17884, this is what I would do with the hatch demo.  No need to review or merge, just for discussion

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by story645** [comment]
this would be a non-starter for me if the reference wasn't the thumbnail 'cause that's what I'd want to be easily findable in the gallery. Also these two examples don't really connect or build on each other - the connecting thread is just the use of hatches.

But going big picture, I think the gallery is a great example of the taxonomy/discoverability problem that's endemic to the docs. I think we have too many examples that are trying to show too many things & end up overlapping in ways that make them redundant while also obfuscating what the example is actually trying to demonstrate and make the current higher level organization for the docs feel a bit scattered. So I'm of the opinion that we need more:
1. smaller byte size "unit test type" examples
2. well constructed with some narrative integration size examples 
3. longer scaffolded narrative tutorials that walk the user from unit test to all the things
and/or it'd be awesome if someone knew how to implement tags on sphinx gallery. 

The too many but all over the place is also why I really think we should be moving towards the gallery as catch all, but also create much more curated sub-galleries for the two things folks probably google most that would be most readily apparent in simple thumbnails:
1. chart types - and these should be like 5 lines max it's just about the plotting call.
2. valid parameter arguments - these are the examples that are really illustrated API doc where it's not even about the code.

File: examples/shapes_and_collections/hatch_demo.py [hunk 1]
```diff
@@ -1,38 +1,69 @@
...
+# ---------------------
+#
+# For convenience we document the 9 different possible hatch styles using the
+# code below.  Note that repeating the hatch character increases the density of
+# of the hatches, as shown in the first row and with "++".  Note also that
+# hatches can be combined ("/+O" in the first row).
+
+fig, axs = plt.subplots(3, 4, constrained_layout=True, figsize=(5, 4))
+
+hatches = ['/', '///', '/+O', '\\',
```
Comment (timhoffm) [inline examples/shapes_and_collections/hatch_demo.py:55]: Not a fan of this. This is a wild mixture of the basic styles and combinations, wich makes it hard to follow.

IMHO we should structure it

- first show the basic building blocks
- explain repetitions
- explain combinations 

Note also that `'\\'` is tricky because it's an escaped backslash not a repeated backslash.

**Comment by jklymak** [comment]
It seems there is a strong consensus to have each gallery entry have only one "purpose".  So I'll close this and leave you guys to it.
